Yosemite/Yosemite/Stores/AppSettingsStore.swift

```diff
@@ -488,11 +488,10 @@ private extension AppSettingsStore {
 
         var dataToSave = preselectedData
 
-        if preselectedData.contains(newPreselectedProvider),
-           let index = preselectedData.firstIndex(of: newPreselectedProvider) {
-            dataToSave[index] = newPreselectedProvider
+        if preselectedData.contains(newPreselectedProvider) {
+            dataToSave[0] = newPreselectedProvider
         } else {
-            dataToSave.append(newPreselectedProvider)
+            dataToSave.insert(newPreselectedProvider, at: 0)
```

Comment:
Here I opted for updating or inserting depending if the `newPreselectedProvider` can be found in the list or not, however, I believe we don't really need to insert it for what we want to accomplish, we could just always update the new value to `dataToSave[0] = newPreselectedProvider`. What do you think?

Reply:
I agree, but would go one further:

`dataToSave = [newPreselectedProvider]`

Shows that there's only ever one, overwrites all the old data in the file, but doesn't break compatibility.